src/securityPolicy.c

```diff
@@ -343,17 +352,28 @@ security_policy_t policyForSignTxOutputAddressParams(
 {
 	DENY_UNLESS(isValidAddressParams(params));
 
-	// network identification should be consistent across tx
-	if (params->type == BYRON) {
+	// address type and network identification
+	switch (params->type) {
+
+	case BYRON:
 		DENY_IF(params->protocolMagic != protocolMagic);
-	} else { // shelley
+		break;
+
+	case REWARD_KEY:
+	case REWARD_SCRIPT:
```

Comment:
These also fall under the determineSpendingChoice, since they result in SPENDING_NONE, is it good to separate them like this? Are reward addresses this much a special case?